common/debug.go

```diff
@@ -0,0 +1,52 @@
+package common
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"os"
+)
+
+func DebugRequest(req *http.Request, rsp *http.Response) {
+	DumpRequest(os.Stdout, req, rsp)
+}
+
+func DumpRequest(writer io.Writer, req *http.Request, rsp *http.Response) {
```

Comment:
does this need to be exported? should this be `dumpRequest`? It cannot be directly used by the caller of the library as a `debug` function anyhow since it doesn't match the expected function signature of `func(req *http.Request, rsp *http.Response)`

Reply:
I went with yes, but it's like a very light yes. I figure, you could use this to dump to a file if you wanted (you'd have to pass in a custom callback function but it could invoke this function with a file handle). But I'm basically indifferent, so I can un-export it if you have even modestly strong feelings about it.

Reply:
I un-exported it. If we need it later, it's simple enough to re-export.